### PR TITLE
[#4091] my resources optimization

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. The query optimization implemented in this PR, will be noticeable in terms of page load time for the 'my-resources' page by users who have large number of owned resources (~200).
2. For the purpose of testing page load improvement, you have to create at least 100 resources. Then compare the page load time for the 'my-resources' page between this branch and develop branch. You should see better page load time with this branch compared to the develop branch.

### CODE REVIEW

File: hs_core/models.py [hunk 1]
```diff
@@ -2085,7 +2085,9 @@ def supports_folders(self):
     @property
     def last_updated(self):
         """Return the last updated date stored in metadata"""
-        return self.metadata.dates.all().filter(type='modified')[0].start_date
+        for dt in self.metadata.dates.all():
```
Comment (pkdash) [inline hs_core/models.py:2088]: This reduces 1 query per resource.

File: hs_core/models.py [hunk 2]
```diff
@@ -4245,7 +4247,7 @@ def resource(self):
     @property
     def title(self):
         """Return the first title object from metadata."""
-        return self._title.all().first()
+        return self._title.all()[0]
```
Comment (pkdash) [inline hs_core/models.py:4250]: This reduces 1 query per resource.

File: hs_core/views/utils.py [hunk 3]
```diff
@@ -666,18 +666,20 @@ def get_my_resources_list(user, annotate=False, filter=None, **kwargs):
             is_favorite=Case(When(short_id__in=favorite_resources.values_list('short_id', flat=True),
                                   then=Value(True, BooleanField()))))
 
-        resource_collection = resource_collection.only('short_id', 'resource_type', 'created')
+        resource_collection = resource_collection.only('short_id', 'resource_type', 'created', 'content_type')
         # we won't hit the DB for each resource to know if it's status is public/private/discoverable
         # etc
-        resource_collection = resource_collection.select_related('raccess', 'rlabels')
+        resource_collection = resource_collection.select_related('raccess', 'rlabels', 'content_type')
```
Comment (pkdash) [inline hs_core/views/utils.py:672]: This reduces 1 query per resource.

File: hs_composite_resource/tests/test_composite_resource.py [hunk 4]
```diff
@@ -3882,33 +3882,75 @@ def _test_zip_file_contents(self, zipfile, aggregation):
...
-    def test_composite_resource_my_resources_scales(self):
-        # test that db queries for "my_resources" remain constant when adding more resources
+    def test_composite_resource_my_resources_one(self):
+        # test that only a fixed number of db queries (based on the number of resources) are
+        # generated for "my_resources" page - this test is with one resource.
 
         # there should not be any resource at this point
         self.assertEqual(BaseResource.objects.count(), 0)
 
+        self.client.login(username='user1', password='mypassword1')
```
Comment (devincowan) [inline hs_composite_resource/tests/test_composite_resource.py:3892]: yep, login is important! :) thanks for fixing this

File: hs_composite_resource/tests/test_composite_resource.py [hunk 5]
```diff
@@ -3969,3 +4011,19 @@ def test_composite_resource_landing_scales(self):
         # accessing the readme file should only be 1 db query
         with self.assertNumQueries(1):
             _ = self.composite_resource.readme_file
+
+    def _get_expected_query_count(self, number_of_resources):
```
Comment (devincowan) [inline hs_composite_resource/tests/test_composite_resource.py:4015]: NICE!

File: theme/templates/includes/my-resources-trows.html [hunk 6]
```diff
@@ -73,8 +73,10 @@
     {% endif %}
     {% endfor %}
   </td>
+   {# QUERY COUNT: this generates (res|get_user_privilege:user) generates 3 queries per resource)  #}
```
Comment (devincowan) [inline theme/templates/includes/my-resources-trows.html:76]: Nit: `generates` twice

**Review by devincowan — APPROVED** [review]
Looks good, one nitpick comment. Thanks for doing this!
I'm waiting to approve until 2.11 release is upon us

File: theme/templates/includes/my-resources-trows.html (hunk 6 above)
Comment (pkdash) [reply theme/templates/includes/my-resources-trows.html]: fixed it.

**Review by devincowan — COMMENTED** [review]
looks good, thanks Pabitra

**Comment by pkdash** [comment]
@devincowan Can you approve again?

**Comment by devincowan** [comment]
@pkdash yep I was waiting till after Keycloak release to approve/merge anything else
But we are going to delay keycloak, so will approve

**Pull request merged**